### PR TITLE
Issue #1709

### LINKED ISSUE #1709: Step 2 'filters' label not always shown
Visible on the systest Portal currently https://portal-systest.aodn.org.au/imos123/home?uuid=8cea53a2-68d6-4d94-a3bf-29d76f14d092&uuid=4402cb50-e20a-44ee-93e6-4728259250d2

**Steps to reproduce**
Click on that link
Look at the filters panel for the Argo layer
Look at the filters panel for the AATAMS layer

**What does happen**
Argo has a 'Filters' label which other filters (not geographic or temporal) come under
AATAMS doesn't have the label
The label is only applied if there are boolean filters configured for a laye

**What should happen**
The 'filters' label should be shown if there are any filters present (other than geographic and temporal)

**Argo (has label)**
has_label

**AATAMS (no label)**
no_label

### PR DESCRIPTION
**Pull request by dnahodil**
Resolves #1709

### CODE REVIEW

File: web-app/js/portal/filter/ui/FilterGroupPanel.js [hunk 1]
```diff
@@ -216,36 +210,36 @@ Portal.filter.ui.FilterGroupPanel = Ext.extend(Ext.Container, {
...
-                currentType = panel.constructor;
+            var newLabel = this._typeLabelForPanel(panel);
+            var newType = panel.constructor;
 
-                currentContainer = this._includeNewGroupContainer(panel);
+            if (newLabel != currentLabel) {
+                this._addLabelToContainer(newLabel, this);
             }
+            else if (newType != currentType) {
+                this.add(this._createVerticalSpacer(15));
```
Comment (dnahodil) [inline web-app/js/portal/filter/ui/FilterGroupPanel.js]: This vertical space should really be done with CSS

File: src/test/javascript/portal/filter/ui/FilterGroupPanelSpec.js [hunk 2]
```diff
@@ -283,8 +285,8 @@ describe("Portal.filter.ui.FilterGroupPanel", function() {
         it('creates a new groups as required', function() {
```
Comment (danfruehauf) [inline src/test/javascript/portal/filter/ui/FilterGroupPanelSpec.js]: Maybe just fix the grammer for this test, not a biggie (I know it's not a line you've changed).

**Comment by danfruehauf** [comment]
Looks fine, I'll just verify it works.

**Comment by danfruehauf** [comment]
Yep, good to go, unless you want to fix that grammer.

**Comment by dnahodil** [comment]
I will fix those test names


University of Tasmania Electronic Communications Policy (December, 2014).
This email is confidential, and is for the intended recipient only. Access, disclosure, copying, distribution, or reliance on any of it by anyone outside the intended recipient organisation is prohibited and may be a criminal offence. Please delete if obtained in error and email confirmation to the sender. The views expressed in this email are not necessarily the views of the University of Tasmania, unless clearly intended otherwise.

**Comment by dnahodil** [comment]
Change made @danfruehauf

**Pull request merged**